Description: Disable old and insecure cyphers in TLS driver
 Disabled:
 * Export ciphers - broken by design, 40 and 56 bit encryption.
 * Low encryption ciphers - 56 and 64 bit encryption.
 * SSLv2 ciphers - some ciphers using MD5 MAC.
 .
 This patch is a backport of changes introduced by the commit
 d2d51381ec3fea97d0bd968cd7ffed2364b644c6 in the upstream Git repository
 to the ejabberd code base as of version 2.1.12.
Author: Janusz Dziemidowicz <rraptorr@nails.eu.org>
Forwarded: not-needed
Last-Update: 2013-09-29
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/tls/tls_drv.c
+++ b/src/tls/tls_drv.c
@@ -44,6 +44,8 @@ typedef unsigned __int32 uint32_t;
 #define SSL_OP_NO_TICKET 0
 #endif
 
+#define CIPHERS "DEFAULT:!EXPORT:!LOW:!SSLv2"
+
 /*
  * R15B changed several driver callbacks to use ErlDrvSizeT and
  * ErlDrvSSizeT typedefs instead of int.
@@ -356,6 +358,8 @@ static ErlDrvSSizeT tls_drv_control(ErlDrvData handle,
 
 	    SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2|SSL_OP_NO_TICKET);
 
+	    SSL_CTX_set_cipher_list(ctx, CIPHERS);
+
 	    SSL_CTX_set_session_cache_mode(ctx, SSL_SESS_CACHE_OFF);
 	    SSL_CTX_set_default_verify_paths(ctx);
 #ifdef SSL_MODE_RELEASE_BUFFERS
